database-migration/src/main/java/uk/ac/ebi/eva/dbmigration/mongodb/ExtractStatisticsFromVariant.java

```diff
@@ -0,0 +1,115 @@
+        logger.info("1) migrate statistics {}", MongoMigrationMain.variantsCollectionName);
+
+        final DBCollection statisticsCollection = mongoTemplate
+                .getCollection(MongoMigrationMain.statisticsCollectionName);
+        final DBCollection variantsCollection = mongoTemplate.getCollection(MongoMigrationMain.variantsCollectionName);
+
+        BulkWriteOperation bulkInsertMaf = variantsCollection.initializeUnorderedBulkOperation();
+        BulkWriteOperation bulkInsertStats = statisticsCollection.initializeUnorderedBulkOperation();
+
+        int counter = 0;
+        DBCursor variantCursor = variantsCollection.find();
+
+        while (variantCursor.hasNext()) {
+            DBObject variantObj = variantCursor.next();
+
+            DBObject statsObj = (DBObject) variantObj.get("st");
+
+            if (statsObj != null) {
+                for (String standaloneStatsObj : statsObj.keySet()) {
+                    BasicDBObject iStatsObj = (BasicDBObject) statsObj.get(standaloneStatsObj);
+
+                    iStatsObj.put("vid", variantObj.get("_id"));
+                    iStatsObj.put("chr", variantObj.get("chr"));
+                    iStatsObj.put("start", variantObj.get("start"));
+                    iStatsObj.put("ref", variantObj.get("ref"));
+                    iStatsObj.put("alt", variantObj.get("alt"));
+
+                    bulkInsertStats.insert(iStatsObj);
+
+                    if (iStatsObj.containsField("maf")) {
+                        DBObject update = new BasicDBObject("$addToSet",
+                                                            new BasicDBObject("maf", iStatsObj.get("maf")));
+                        DBObject find = new BasicDBObject("_id", variantObj.get("_id"));
+                        bulkInsertMaf.find(find).updateOne(update);
+                    }
+
+                    counter++;
+                    if (counter % BULK_SIZE == 0) {
+                        bulkInsertStats.execute();
+                        bulkInsertStats = statisticsCollection.initializeUnorderedBulkOperation();
+                        bulkInsertMaf.execute();
+                        bulkInsertMaf = variantsCollection.initializeUnorderedBulkOperation();
+                    }
+                }
+            }
+
+        }
+
+        if (counter % BULK_SIZE != 0) {
+            bulkInsertStats.execute();
+            bulkInsertMaf.execute();
+        }
+
+        //before executing the next changeSet check that the count of read and written statistics documents match
+        if (counter != statisticsCollection.count()) {
+            logger.error(
+                    "The number of processed Variants ({}) is different from the number of new statistics inserted ({})." +
+                            " The 'st' field will not be removed from the {} collection.",
+                    counter, statisticsCollection.count(), MongoMigrationMain.variantsCollectionName);
+            System.exit(1);
```

Comment:
`Exit` replaced with a `RuntimeException` in this way the `mongobeelock` is released. If this check fails then the statistics collection is populated, the 'maf' field is added into the Variant but the 'st' field is not removed from Variant. The `dbchangelog` and `mongobeelock` are empty. If we re-run the procedure statistics are re-inserted :s